app/src/main/java/de/danoeh/antennapod/fragment/preferences/SynchronizationPreferencesFragment.java

```diff
@@ -0,0 +1,294 @@
+        }
+    }
+
+    private void setupScreen() {
+        final Activity activity = getActivity();
+
+        findPreference(PREFERENCE_LOGIN).setOnPreferenceClickListener(preference -> {
+            AlertDialog.Builder builder = new AlertDialog.Builder(getContext());
+            builder.setTitle(R.string.dialog_choose_sync_service_title);
+            String[] syncProviderDescriptions = {
+                    getString(R.string.gpodnet_description),
+                    getString(R.string.preference_synchronization_summary_nextcloud)
+            };
+            int[] syncProviderIcons = {R.drawable.gpodder_icon, R.drawable.nextcloud_logo_svg};
+
+            ListAdapter adapter = new ArrayAdapter<String>(
+                    getContext(), R.layout.alertdialog_sync_provider_chooser, syncProviderDescriptions) {
+
+                ViewHolder holder;
+
+                class ViewHolder {
+                    ImageView icon;
+                    TextView title;
+                }
+
+                public View getView(int position, View convertView, ViewGroup parent) {
+                    final LayoutInflater inflater = LayoutInflater.from(getContext());
+                    if (convertView == null) {
+                        convertView = inflater.inflate(
+                                R.layout.alertdialog_sync_provider_chooser, null);
+
+                        holder = new ViewHolder();
+                        holder.icon = (ImageView) convertView
+                                .findViewById(R.id.icon);
+                        holder.title = (TextView) convertView
+                                .findViewById(R.id.title);
+                        convertView.setTag(holder);
+                    } else {
+                        holder = (ViewHolder) convertView.getTag();
+                    }
+
+                    holder.title.setText(syncProviderDescriptions[position]);
+
+                    holder.icon.setImageResource(syncProviderIcons[position]);
+                    return convertView;
+                }
+            };
+
+            builder.setAdapter(adapter, (dialog, which) -> {
+                String userSelect = SYNC_PROVIDER_CHOICE_GPODDER_NET;
+                switch (which) {
+                    case 0:
+                        break;
+                    case 1:
+                        userSelect = SyncService.SYNC_PROVIDER_CHOICE_NEXTCLOUD;
+                        break;
+                    default:
+                        break;
+                }
+                setSelectedSyncProvider(userSelect);
+                if (userSelect.equals(SYNC_PROVIDER_CHOICE_GPODDER_NET)) {
+                    new GpodderAuthenticationFragment()
+                            .show(getChildFragmentManager(), GpodderAuthenticationFragment.TAG);
+                    updateScreen();
+                    return;
+                }
+                openNextcloudAccountChooser();
```

Comment:
I think it would be more clean to do this inside the switch statement. Currently you check twice which provider is selected.

Reply:
totally :-)